sl1m/qp.py

```diff
@@ -13,55 +27,290 @@ def quadprog_solve_qp(P, q, G=None, h=None, C=None, d=None, verbose = False):
     qp_a = -q
     if C is not None:
         if G is not None:
-                qp_C = -vstack([C, G]).T
-                qp_b = -hstack([d, h])   
+            qp_C = -vstack([C, G]).T
+            qp_b = -hstack([d, h])
         else:
-                qp_C = -C.transpose()
-                qp_b = -d 
+            qp_C = -C.transpose()
+            qp_b = -d
         meq = C.shape[0]
-    else:  # no equality constraint 
+    else:  # no equality constraint
         qp_C = -G.T
         qp_b = -h
-        meq = 0 
+        meq = 0
     res = quadprog.solve_qp(qp_G, qp_a, qp_C, qp_b, meq)
     if verbose:
-            return res
+        return res
     #~ print 'qp status ', res
     return res[0]
 
 
-#min ||Ax-b||**2 
+#min ||Ax-b||**2
 #subject to  G x <= h
 #subject to  C x  = d
 def solve_least_square(A,b,G=None, h=None, C=None, d=None):
-        P = dot(A.T, A)
-        #~ q = 2*dot(b, A).reshape(b.shape[0])
-        q = -dot(A.T, b).reshape(b.shape[0])
-        #~ q = 2*dot(b, A) 
-        return quadprog_solve_qp(P, q, G, h, C, d)
+    P = dot(A.T, A)
+    #~ q = 2*dot(b, A).reshape(b.shape[0])
+    q = -dot(A.T, b).reshape(b.shape[0])
+    #~ q = 2*dot(b, A)
+    return solve_qp_quadprog(P, q, G, h, C, d)
+
+#min x' C x + c x
```

Comment:
This should be a docstring for the function

Reply:
this file should have been deleted sorry